### PR TITLE
PR: Remove redundant variables

### PR DESCRIPTION
**Pull request by rlaverde**
Remove `project_name` and `repo_name`, and use instead `plugin_name` modifications.

### CODE REVIEW

File: spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-') }}/circle.yml [hunk 1]
```diff
@@ -43,11 +43,11 @@ dependencies:
 test:
   override:
     # Check Python style
-    - $CMD_LINT_CIOCHECK {{ cookiecutter.repo_name }} -dt: # note the colon
+    - $CMD_LINT_CIOCHECK spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '_') }} -dt: # note the colon
```
Comment (goanpeca) [inline spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-') }}/circle.yml]: there is an error there should be _ not -

File: spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-') }}/circle.yml [hunk 2]
```diff
@@ -43,11 +43,11 @@ dependencies:
...
         parallel: true
 
     # Run Python tests
-    - $CMD_TEST_PYTEST {{ cookiecutter.repo_name }} --cov={{ cookiecutter.repo_name }}: # note the colon        parallel: true
+    - $CMD_TEST_PYTEST spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '_') }} --cov=spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '_') }}: # note the colon        parallel: true
```
Comment (goanpeca) [inline spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-') }}/circle.yml]: same here

**Review by goanpeca — CHANGES_REQUESTED** [review]

**Comment by rlaverde** [comment]
I'm not sure if this is a good change, directory and file names become really long and ugly .___.

I'm closing this, I think that's not to much harm to be asked two more questions, that will be ok to leave with the default value. Maybe we should add a comment in the Readme instead.

**Comment by goanpeca** [comment]
>I'm not sure if this is a good change, directory and file names become really long and ugly

I don't understand how this is a problem? unless it does not work on windows or something is not enough reason

**Comment by rlaverde** [comment]
I address your comments, anyway I dont like to have files named:

`spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-') }}/spyder_{{ cookiecutter.plugin_name.lower().replace(' ', '_') }}}/widgets/{{ cookiecutter.plugin_name.lower().replace(' ', '') }}gui.py`

and when writing in the terminal with escaping will become:

`spyder-\{\{\ cookiecutter.plugin_name.lower\(\).replace\(\'\ \'\,\ \'-\'\)\ \}\}/spyder_\{\{\ cookiecutter.plugin_name.lower\(\).replace\(\'\ \'\,\ \'_\'\)\ \}\}/widgets/\{\{\ cookiecutter.plugin_name.lower\(\).replace\(\'\ \'\,\ \'\'\)\ \}\}gui.py`

I think this is so error prone :confused:

File: spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-') }}/circle.yml [hunk 3]
```diff
@@ -43,11 +43,11 @@ dependencies:
 test:
   override:
     # Check Python style
-    - $CMD_LINT_CIOCHECK spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '_') }} -dt: # note the colon
+    - $CMD_LINT_CIOCHECK spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-') }} -dt: # note the colon
```
Comment (goanpeca) [inline spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-') }}/circle.yml]: This is wrong we are calling the module here

File: spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-') }}/circle.yml [hunk 4]
```diff
@@ -43,11 +43,11 @@ dependencies:
...
         parallel: true
 
     # Run Python tests
-    - $CMD_TEST_PYTEST spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '_') }} --cov=spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '_') }}: # note the colon        parallel: true
+    - $CMD_TEST_PYTEST spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-') }} --cov=spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-') }}: # note the colon        parallel: true
```
Comment (goanpeca) [inline spyder-{{ cookiecutter.plugin_name.lower().replace(' ', '-') }}/circle.yml:51]: same here

**Review by goanpeca — CHANGES_REQUESTED** [review]

**Review by goanpeca — APPROVED** [review]

**Comment by rlaverde** [comment]
Closing, I don't like the approach of this PR, I think is better to continue asking for 3 variables.

Maybe this change upstream https://github.com/audreyr/cookiecutter/issues/364 will make this a lot easier

**Pull request closed without merging**